ddtrace/debugging/_probe/remoteconfig.py

```diff
@@ -287,27 +307,20 @@ def _update_probes_for_config(self, config_id, config):
         next_probes = (
             {probe.probe_id: probe for probe in get_probes(config_id, config)} if config not in (None, False) else {}
         )  # type: Dict[str, Probe]
-
+        log.debug("Dynamic Instrumentation, dispatch probe events")
         self._dispatch_probe_events(prev_probes, next_probes)
 
         if next_probes:
             self._configs[config_id] = next_probes
         else:
             self._configs.pop(config_id, None)
 
-    def __call__(self, metadata, config):
-        # type: (Optional[ConfigMetadata], Any) -> None
-
-        # DEV: We emit a status update event here to avoid having to spawn a
-        # separate thread for this.
-        if time.time() > self._status_timestamp:
-            log.debug("Emitting probe status log messages")
-            probes = [probe for config in self._configs.values() for probe in config.values()]
-            self._callback(ProbePollerEvent.STATUS_UPDATE, probes)
-            self._next_status_update_timestamp()
 
-        if metadata is None:
-            log.debug("no RCM metadata")
-            return
+class ProbeRCAdapter(PubSub):
+    __publisher_class__ = RemoteConfigPublisher
+    __subscriber_class__ = DebuggerRemoteConfigSubscriber
+    __shared_data = PublisherSubscriberConnector()
 
-        self._update_probes_for_config(metadata.id, config)
+    def __init__(self, _preprocess_results, callback):
```

Comment:
what is `_preprocess_results` for?

Reply:
It's a function that Publisher instance executes in the main process:
https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/internal/remoteconfig/_publishers.py#L106

This function callback has the responsibility of:
- Transforming payloads
- Executing actions ONLY in the main process

Example: ASM needs to register some Remote Config products dynamically in the main process because the child process doesn't care about products.
https://github.com/DataDog/dd-trace-py/pull/5464/files#diff-6c9b0db1351f5cf53547609e99457bf2d1480ba0e8e123ec9be608aeef4712d5R119

Reply:
Ah I see. But do I understand it right that for the debugger we are passing an empty string to it?
```
di_callback = self.__rc_adapter__("", self._on_configuration)
```

Reply:
ouch, it works correctly with a empty string, but the correct value should be `None`, good point @P403n1x87 !